src/basic/string-util.c

```diff
@@ -1069,3 +1069,35 @@ bool string_is_safe(const char *p) {
 
         return true;
 }
+
+int string_isvalid_interval(const char *instr) {
+        char *pstr  = (char *)instr;  /* tmp */
+        char *pstr_front  = pstr;  /* front char */
+        char *pstr_behind = pstr;  /* behind char */
+
+        if (isempty(instr))
+                return 1;
+
+        while (*pstr != '\0') {
+                /* behind */
+                pstr_behind = pstr + 1;
+
+                /* 0-3,4,6,7-10 */
+                if (((*pstr < '0') || (*pstr > '9')) &&
+                    (*pstr != '-') && (*pstr != ','))
+                        return 2;
+
+                /* - , must is a num */
+                if (('-' == *pstr) || (',' == *pstr)) {
+                        if ((*pstr_front < '0') || (*pstr_front > '9') ||
+                            (*pstr_behind < '0') || (*pstr_behind > '9'))
```

Comment:
Coding style: please do not use () around comparisons. Comparison operators have higher priority than logical boolean operators on purpose, and those parens add a lot of noise.

That said, is that function even necessary? It seems that just reusing the parsing function is enough — it should either return a value or an error. Writing a separate checker function is then not necessary.